transformation_test/testdata/logging_processor-parse_regex_optional_groups/config.yaml

```diff
@@ -0,0 +1,5 @@
+- type: parse_regex
+  field: message
+  regex: ^\<(?<pri>[0-9]{1,5})\>1 (?<time>[^ ]+) (?<host>host.*)? (?<ident>[^ ]+) (?<pid>[n0-9]+) (?<msgid>[^ ]+) (?<extradata>.*) (?<message>.*)$
```

Comment:
Can we make the test cases simpler to reduce the extraneous fields? I think all three cases can be tested with

```yaml
regex: ^(?<always>\S+) (?<empty>\S*) (?<optional>\S+)? (?<name>.+)$
```
and this test log:

```
first second third all fields present
first second  optional field missing
first  third empty field missing
first   optional and empty fields missing
   always, optional and empty missing
```